src/Framework/TestCase.php

```diff
@@ -1942,89 +1946,100 @@ private function checkRequirements(): void
 
     private function handleDependencies(): bool
     {
-        if (!empty($this->dependencies) && !$this->inIsolation) {
-            $className  = \get_class($this);
-            $passed     = $this->result->passed();
-            $passedKeys = \array_keys($passed);
+        if (empty($this->dependencies) || $this->inIsolation) {
+            return true;
+        }
 
-            foreach ($passedKeys as $key => $value) {
-                $pos = \strpos($value, ' with data set');
+        $className  = \get_class($this);
+        $passed     = $this->result->passed();
+        $passedKeys = \array_keys($passed);
+        $numKeys    = \count($passedKeys);
 
-                if ($pos !== false) {
-                    $passedKeys[$key] = \substr($value, 0, $pos);
-                }
+        for ($i = 0; $i < $numKeys; $i++) {
+            $pos = \strpos($passedKeys[$i], ' with data set');
+
+            if ($pos !== false) {
+                $passedKeys[$i] = \substr($passedKeys[$i], 0, $pos);
             }
+        }
 
-            $passedKeys = \array_flip(\array_unique($passedKeys));
+        $passedKeys = \array_flip(\array_unique($passedKeys));
 
-            foreach ($this->dependencies as $dependency) {
-                $deepClone    = false;
-                $shallowClone = false;
+        foreach ($this->dependencies as $dependency) {
+            if (empty($dependency)) {
```

Comment:
Since we touch this, what do you think about using something that is a bit more explicit instead of `empty()`?

Reply:
I like it, will change